projects/angular-calendar/src/modules/common/keydown.directive.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  Directive,
+  Renderer2,
+  ElementRef,
+  OnInit,
+  OnDestroy,
+  Output,
+  EventEmitter,
+  Inject,
+  Input,
+  NgZone
+} from '@angular/core';
+import { DOCUMENT } from '@angular/common';
+import { Observable, Subject } from 'rxjs';
+import { requestIdleCallbackObservable } from './request-idle-callback';
+import { switchMapTo, takeUntil } from 'rxjs/operators';
+
+const keydownElements = new Set<HTMLElement>();
+
+const eventName = 'keydown.enter';
+
+@Directive({
+  selector: '[mwlKeydown]'
+})
+export class KeydownDirective implements OnInit, OnDestroy {
+  @Input() keydownListenerDisabled = false;
+
+  @Output('mwlKeydown') keydown = new EventEmitter<KeyboardEvent>(); // tslint:disable-line
+
+  private destroy$ = new Subject();
+
+  constructor(
+    private renderer: Renderer2,
+    private elm: ElementRef<HTMLElement>,
+    @Inject(DOCUMENT) private document,
+    private zone: NgZone
+  ) {}
+
+  ngOnInit(): void {
+    if (!this.keydownListenerDisabled) {
+      this.renderer.setAttribute(
+        this.elm.nativeElement,
+        'data-calendar-keyable',
+        'true'
+      );
+      keydownElements.add(this.elm.nativeElement);
+
+      requestIdleCallbackObservable()
+        .pipe(
+          switchMapTo(this.listen()),
+          takeUntil(this.destroy$)
+        )
+        .subscribe(event => {
+          // prevent default override of the enter key
+          if (
+            event.keyCode === 13 ||
+            event.which === 13 ||
+            event.key === 'Enter'
+          ) {
+            event.preventDefault();
+          }
+          // prevent child click events from firing on parent elements that also have click events
+          let nearestClickableParent = event.target as HTMLElement;
```

Comment:
This was also a hack needed for hammerjs, `event.stopPropagation();` should be enough, rather than manually doing this traversal of the dom